egs/wsj/s5/steps/dict/apply_g2p_phonetisaurus.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+# Copyright 2014  Johns Hopkins University (Author: Yenda Trmal)
+# Copyright 2016  Xiaohui Zhang
+#           2018  Ruizhe Huang
+# Apache 2.0
+
+# This script applies a trained Phonetisarus G2P model to
+# synthesize pronunciations for missing words (i.e., words in
+# transcripts but not the lexicon), and output the expanded lexicon.
+# The user could specify either nbest or pmass option 
+# to determine the number of output pronunciation variants, 
+# or use them together to get the intersection of two options.
+
+# Begin configuration section.  
+stage=0
+nbest=          # Generate up to $nbest variants
+pmass=          # Generate so many variants to produce $pmass ammount, like 90%, of the prob mass
+# End configuration section.
+
+echo "$0 $@"  # Print the command line for logging
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. utils/parse_options.sh || exit 1;
+
+set -u
+set -e
+
+if [ $# != 3 ]; then
+  echo "Usage: $0 [options] <g2p-model> <word-list> <lexicon-out>"
+  echo "... where <g2p-model> is the trained g2p model."
+  echo "          <word-list> is a list of words whose pronunciation is to be generated."
+  echo "          <lexicon-out> output lexicon, whose format is <word>\t<prob>\t<pronunciation> for each line."
+  echo "e.g.: $0 --nbest 1 exp/g2p/model.fst exp/g2p/oov_words.txt data/local/dict_nosp/lexicon.txt"
+  echo ""
+  echo "main options (for others, see top of script file)"
+  echo "  --nbest <int>    # Maximum number of hypotheses to produce. By default, nbest=20"
+  echo "  --pmass <float>  # Select the maximum number of hypotheses summing to a total mass of pmass amount, within [0, 1], for a word. By default, pmass=1.0"
+  exit 1;
+fi
+
+model=$1
+word_list=$2
+out_lexicon=$3
+out_lexicon_failed="${out_lexicon}.failed"
```

Comment:
Also, to keep the convention, probably you should ask the user to specify $outdir and write the output lexicon as $outdir/lexicon.lex as was done in the current apply_g2p.sh. And then put the list of failed words as $outdir/lexicon.failed

Reply:
Fixed. Many thanks for all the above suggestions!